src/containers/Transaction/Transaction.js

```diff
@@ -0,0 +1,174 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useForm, Controller } from 'react-hook-form';
+import * as Component from '@material-ui/core';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import styles from './transaction.module.scss';
+import Text from '../../components/Text';
+
+const defaultValues = {
+  isBuying: true,
+  currencyName: '',
+  price: '',
+  amount: 0,
+  total: ''
+};
+
+const schema = yup.object().shape({
+  currencyName: yup
+    .string()
+    .required(),
+  amount: yup
+    .number()
+    .positive()
+    .integer()
+    .transform((val) => (Number.isNaN(val) ? undefined : val))
+    .required()
+});
+
+export const errorMessages = {
+  required: 'The field is required',
+  typeError: 'Value must be a number',
+  integer: 'Value must be an integer',
+  min: 'Amount must be a positive number'
+};
+
+const Transaction = ({ stockExchangeData }) => {
+  const {
+    reset,
+    control,
+    handleSubmit,
+    watch,
+    setValue,
+    getValues,
+    formState: { errors }
+  } = useForm({
+    defaultValues,
+    resolver: yupResolver(schema)
+  });
+  const [isBuying, currencyName, amount] = [watch('isBuying'), watch('currencyName'), watch('amount')];
+
+  const onSubmit = (data) => {
+    // eslint-disable-next-line no-console
+    console.log(data);
+  };
+
+  useEffect(() => {
+    setValue('price', stockExchangeData.find((dataItem) => dataItem.name === currencyName)?.price || '');
+  }, [currencyName, setValue, stockExchangeData]);
+
+  useEffect(() => {
+    if (amount && currencyName && !errors.amount) {
+      const total = stockExchangeData.find((dataItem) => dataItem.name === currencyName).price * getValues('amount');
+      setValue('total', total.toFixed(2));
```

Comment:
We have some inconsistency in interface in general:
![Zrzut ekranu 2021-05-10 o 17 41 16](https://user-images.githubusercontent.com/17573378/117686822-85643300-b1b7-11eb-8ea1-430c43f36511.png)

All those number values could be numbers in fact right?